packages/trpc/server/routers/viewer/filterSegments/preference.handler.ts

```diff
@@ -0,0 +1,46 @@
+import { prisma } from "@calcom/prisma";
+import type { TrpcSessionUser } from "@calcom/trpc/server/types";
+
+import type { TSetFilterSegmentPreferenceInputSchema } from "./preference.schema";
+
+export const setPreferenceHandler = async ({
```

Comment:
@eunjae-lee 
Can we create "packages/lib/server/repository/userFilterSegmentPreference.ts", create UserFilterSegmentPreferenceRepository, and move this function as a method to that repository? The motivation behind this is to directly use the function in RSCs instead of creating a server trpc caller in the future.

Reply:
@hbjORbj good point. addressed! I named it SegmentFilterRepository, as it's a part of it and I plan to move more to it.